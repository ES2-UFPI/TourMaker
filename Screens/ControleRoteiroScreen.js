import React, { Component } from 'react';
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
import { Table, TableWrapper, Row, Cell } from 'react-native-table-component';
import CustomButton from './ScreensModules/CustomButton';

class ControleRoteiroScreen extends Component{
    constructor(props){
        super(props)
        var table = []
        if (props.route.params.listaPDI != null){
            props.route.params.listaPDI.forEach(element => {
                table.push([element.placeName,element.rating," "])
            });
        }
        this.state ={
            listaPDI: props.route.params.listaPDI,
            tableHead: ["Parada", "Avaliação", 'Excluir'],//avaliação Maps
            tableData: table
        }
    }

    deleteStop(index) { 
        var a = this.state.listaPDI.filter( (item, b) => {
            return b !== index
        })
        this.setState({
            listaPDI: a
        })

        a = this.state.tableData.filter((item, b) => {
            return b !== index
        })
        this.setState({
            tableData: a
        })
        // Refazer/Reorganizar Roteiro
    }

    render(){
        const element = (data, index) => (
            <TouchableOpacity onPress={() => this.deleteStop(index)}>
              <View style={styles.btn}>
                <Text style={styles.btnText}>Excluir</Text>
              </View>
            </TouchableOpacity>
          );

        return(
            <View style={styles.RotaScreen}>
                <Text>Gereciamento de Roteiro</Text>
                <View style={styles.container}>
                    <Table borderStyle={{ borderColor: 'transparent' }}>
                        <Row data={this.state.tableHead} style={styles.head} textStyle={styles.text} />
                        {
                            this.state.tableData.map((rowData, index) => (
                                <TableWrapper key={index} style={styles.row}>
                                    {
                                        rowData.map((cellData, cellIndex) => (
                                            <Cell key={cellIndex} data={cellIndex === 2 ? element(cellData, index) : cellData} textStyle={styles.text}/>
                                        ))
                                    }
                                </TableWrapper>
                            ))
                        }
                    </Table>
                    
                </View>
                <CustomButton
                    title="Adicionar Parada"
                    color={styles.Buttons.color}
                    onPress={()=>{
                        const paramsRota = {
                            listaPDI: this.state.listaPDI
                        }
                        this.props.navigation.push('RoteiroManual', paramsRota)
                    }}
                />
                
            </View>

            
        )
    }
}

const styles = StyleSheet.create({
    RotaScreen: {
      flex: 1,
      backgroundColor: '#fff',
      alignItems: 'center',
      justifyContent: 'flex-start',
    },
    Buttons:{
        color: '#1abc9c'
    },
    container: { flex: 1, padding: 16, paddingTop: 30, backgroundColor: '#fff' },
    head: { height: 40,width: 400, backgroundColor: '#1abc9c' },
    text: { margin: 6 },
    row: { flexDirection: 'row', backgroundColor: '#f1f8ff' },
    btn: { width: 58, height: 18, backgroundColor: '#78B7BB',  borderRadius: 2 },
    btnText: { textAlign: 'center', color: '#fff' } 
  });
export default ControleRoteiroScreen;

