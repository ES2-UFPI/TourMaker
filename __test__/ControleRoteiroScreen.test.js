import React from 'react';
import renderer from 'react-test-renderer';

import ControleRoteiroScreen from '../Screens/ControleRoteiroScreen';

it('renders correctly', () => {
    const tree = renderer.create(<ControleRoteiroScreen route={{
        params:{
            listaPDI: []
        }
    }} />).toJSON();
    expect(tree).toMatchSnapshot();
});