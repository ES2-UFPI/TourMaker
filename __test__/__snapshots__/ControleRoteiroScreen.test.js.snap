// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Text>
    Gereciamento de Roteiro
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "padding": 16,
        "paddingTop": 30,
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "borderBottomWidth": 0,
            "borderColor": "transparent",
            "borderLeftWidth": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            undefined,
            0,
            Object {
              "flexDirection": "row",
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#1abc9d",
              "width": 400,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "transparent",
                "borderRightWidth": 0,
                "borderTopWidth": 0,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            Parada
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "transparent",
                "borderRightWidth": 0,
                "borderTopWidth": 0,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            Avaliação
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "transparent",
                "borderRightWidth": 0,
                "borderTopWidth": 0,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            Excluir
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "transparent",
                "borderRightWidth": 0,
                "borderTopWidth": 0,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            Detalhes
          </Text>
        </View>
      </View>
      <RCTScrollView>
        <View />
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "padding": 16,
        "paddingTop": 30,
      }
    }
  >
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#E3E6E7",
                  "borderRadius": 10,
                  "marginLeft": 30,
                  "marginRight": 30,
                  "minWidth": 300,
                },
                Object {},
              ]
            }
          >
            <View
              style={
                Object {
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "marginBottom": 5,
                    "marginTop": 10,
                    "textAlign": "center",
                  }
                }
              >
                Favorito
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "marginBottom": 10,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Salvar Roteiro Como
              </Text>
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                autoCorrect={true}
                autoFocus={true}
                clearButtonMode="never"
                clearTextOnFocus={false}
                keyboardType="default"
                maxLength={null}
                onChangeText={[Function]}
                onKeyPress={[Function]}
                placeholder="Nome Roteiro"
                rejectResponderTermination={true}
                secureTextEntry={false}
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#B0B0B0",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "color": "rgba(0,0,0,0.54)",
                    "fontSize": 16,
                    "marginBottom": 15,
                    "marginTop": 10,
                    "paddingBottom": 5,
                    "paddingLeft": 10,
                    "paddingTop": 5,
                    "textAlign": "left",
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Object {
                  "borderColor": "#B0B0B0",
                  "borderTopWidth": 1,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "maxHeight": 48,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "borderColor": "#B0B0B0",
                      "borderRightWidth": 5,
                      "color": "#408AE2",
                      "fontSize": 18,
                      "height": 48,
                      "maxHeight": 48,
                      "padding": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  Cancelar
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#B0B0B0",
                    "width": 1,
                  }
                }
              />
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#408AE2",
                      "fontSize": 18,
                      "padding": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  Salvar
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#1abc9c",
                },
              ]
            }
          >
            Salvar Roteiro
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#1abc9c",
                },
              ]
            }
          >
            Adicionar Parada
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
